axum/src/extract/host.rs

```diff
@@ -52,10 +56,28 @@ where
     }
 }
 
+#[allow(warnings)]
+fn parse_forwarded(headers: &HeaderMap) -> Option<&str> {
+    // if there are multiple `Forwarded` `HeaderMap::get` will return the first one
+    let forwarded_values = headers.get(FORWARDED)?.to_str().ok()?;
+
+    // get the first set of values
+    let first_value = forwarded_values.split(',').nth(0)?;
+
+    // find the value of the `for` field
+    first_value.split(';').find_map(|pair| {
+        let (key, value) = pair.split_once('=')?;
+        key.trim().eq_ignore_ascii_case("for").then(|| {
```

Comment:
This extracts out the IP address, not the host. Probably meant to search for `host`. Last key may not have the original host, may be another hop out.

Reply:
Ah yeah you're right. I mistook `x-forwarded-for` as `x-forwarded-host`.

Reply:
In my example above, the last hop only includes a `proto` key, but not a `host`. Not all proxies will necessarily provide all forwarded values.